src/main/java/org/apache/commons/crypto/utils/ReflectionUtils.java

```diff
@@ -73,9 +81,16 @@ public static Class<?> getClassByName(final String name) throws ClassNotFoundExc
      * couldn't be loaded. This is to avoid the overhead of creating an exception.
      *
      * @param name the class name.
-     * @return the class object, or {@code null} if it could not be found.
+     * @return the class object, or {@code null} if it could not be found or initialization failed.
      */
     private static Class<?> getClassByNameOrNull(final String name) {
+        final Set<String> set =
+            INIT_ERROR_CLASSES.computeIfAbsent(CLASSLOADER, k -> Collections.synchronizedSet(new HashSet<>()));
+
+        if (set.contains(name)) {
+            return null;
+        }
```

Comment:
Add a collection to record classes that have `ExceptionInInitializerError` to avoid unnecessary reentry of the `Class.forName` method in this scenario. This collection cannot be garbage collected, otherwise there will still be reentry issues.